content/browser/renderer_host/brave_navigation_controller_impl.h

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CONTENT_BROWSER_RENDERER_HOST_BRAVE_NAVIGATION_CONTROLLER_IMPL_H_
+#define BRAVE_CONTENT_BROWSER_RENDERER_HOST_BRAVE_NAVIGATION_CONTROLLER_IMPL_H_
+
+#include "content/browser/renderer_host/navigation_controller_impl.h"
+#include "content/public/browser/navigation_controller.h"
+
+namespace content {
+class BrowserContext;
+class FrameTree;
+class NavigationControllerDelegate;
+
+class CONTENT_EXPORT BraveNavigationControllerImpl
```

Comment:
Do you think it would help to have some comment above explaining the role of `BraveNavigationControllerImpl`, as opposed to the class it is overrding.

Reply:
thanks, fixed at https://github.com/brave/brave-core/pull/20996/commits/f840e5dc3a8a356859d6cd721ee8345ad7f19a85#diff-22e6afdf9bd6b125060d65f811409fd5c8a4a3b72b7c48ab946d83adc76bc90fR17